pallets/utxo/src/lib.rs

```diff
@@ -841,10 +820,31 @@ pub mod pallet {
             value: Value,
             address: Vec<u8>,
         ) -> DispatchResultWithPostInfo {
-            use chainscript::Script;
-
+            let (_, data, _) = bech32::decode(&address).or_else(|e| match e {
+                bech32::Error::InvalidLength => Err(DispatchError::Other(
+                    "Failed to decode address: invalid length",
+                )),
+                bech32::Error::InvalidChar(_) => Err(DispatchError::Other(
+                    "Failed to decode address: invalid character",
+                )),
+                bech32::Error::MixedCase => {
+                    Err(DispatchError::Other("Failed to decode address: mixed case"))
+                }
+                bech32::Error::InvalidChecksum => Err(DispatchError::Other(
+                    "Failed to decode address: invalid checksum",
+                )),
+                _ => Err(DispatchError::Other("Failed to decode address")),
+            })?;
+
+            let mut tmp: &[u8] = &Vec::<u8>::from_base32(&data)
+                .or_else(|_| Err(DispatchError::Other("Base32 conversion failed")))?;
```

Comment:
It feels to me that the users of the bech32 library should not have to deal with getting data out in two steps. I'd expect there to be a single higher-level call that goes from address directly to `Vec<u8>` data and the base32 conversion would be just an implementation detail of the bech32 libaray.

Also should HRP be checked here to be `ml` or `tm` depending on whether the node is running on testnet or mainnet and rejected otherwise?

Reply:
Sure I don't insist on the HRP checking being done in this PR but I think it should be checked eventually.